src/store/message-store-level.ts

```diff
@@ -87,46 +90,52 @@ export class MessageStoreLevel implements MessageStore {
     const messages: BaseMessage[] = [];
 
     // parse criteria into a query that is compatible with the indexing DB (search-index) we're using
-    const exactTerms = MessageStoreLevel.buildExactQueryTerms(exactCriteria);
+    const exactTerms = MessageStoreLevel.buildExactQueryTerms({ ...exactCriteria, tenant });
```

Comment:
With this change, I think we can/should remove the explicit indexing of tenant in everywhere `put()` is invoked:

* `.../handlers/protocols-configure.ts` - line 63
* `.../handlers/records-delete.ts` - line 94
* `.../handlers/records-write.ts` - line 125
* `.../handlers/permissions-request.ts` - line 33

I'd attempt to make the changes, but also don't want to screw up your workflow.